File: anagrams.js
const anagrams = (s1, s2) =>{
    const count = {}
    for ( let i of s1){
        if(!char in count){
            count[s1[char]] = 1
        }
        count[s1[char]] += 1
    }

    for ( let i of s2){
        if(count[s2[i]] == undefined){
            return false;
        }else{
            count[s2[i]] -= 1;
        }
    }

    for(let i in count){
        if(count[i] != 0){
            return false;
        }
    }
    
    return true

}

console.log(anagrams('cats', 'tocs'))

File: anagrams.py
def anagrams(s1, s2):
  return char_count(s1) == char_count(s2)
  
def char_count(s):
  count = {}

  for char in s:
    if char not in count:
      count[char] = 1
    count[char] += 1
  return count

from collections import Counter

def anagrams2(s1, s2):
  return Counter(s1) == Counter(s2)



print(anagrams('monkeyswrite', 'newyorktimes'))
print(anagrams2('tax', 'taxi'))